internal/service/emr/studio_session_mapping.go

```diff
@@ -0,0 +1,162 @@
+package emr
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/emr"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/tfresource"
+	"github.com/hashicorp/terraform-provider-aws/internal/verify"
+)
+
+func ResourceStudioSessionMapping() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceStudioSessionMappingCreate,
+		Read:   resourceStudioSessionMappingRead,
+		Update: resourceStudioSessionMappingUpdate,
+		Delete: resourceStudioSessionMappingDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"identity_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"identity_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"identity_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(emr.IdentityType_Values(), false),
+			},
+			"session_policy_arn": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"studio_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceStudioSessionMappingCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).EMRConn
+
+	var id string
+	studioId := d.Get("studio_id").(string)
+	indetityType := d.Get("identity_type").(string)
+	input := &emr.CreateStudioSessionMappingInput{
+		IdentityType:     aws.String(indetityType),
+		SessionPolicyArn: aws.String(d.Get("session_policy_arn").(string)),
+		StudioId:         aws.String(studioId),
+	}
+
+	if v, ok := d.GetOk("identity_id"); ok {
+		input.IdentityId = aws.String(v.(string))
+		id = v.(string)
+	}
+
+	if v, ok := d.GetOk("identity_name"); ok {
+		input.IdentityName = aws.String(v.(string))
+		id = v.(string)
+	}
```

Comment:
`id` is set in both of these blocks - Are `identity_id` and `identity_name` mutually exclusive (and one is required)?
Can capture via `ExactlyOneOf` on the attributes.

Reply:
the get method returns both of them ill try again but i think it makes this validations fails

Reply:
added, works. not sure what i did wrong before